### PR TITLE
Generate calico manifests from source

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What type of PR is this?**

**What this PR does / why we need it**: Instead of keeping a static Calico manifests in the repo, this PR uses the Kustomize feature to pull a base manifest from a url to get the vxlan manifest from https://docs.projectcalico.org/v3.20/manifests/calico-vxlan.yaml and makes a few patches necessary for Azure and IPv6. This will make sure we don't drift from the recommended Calico configuration and will make updating the Calico version a lot easier.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
partly addresses #1917 

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
Generate calico manifests from source
```

### CODE REVIEW

File: templates/addons/calico-ipv6/patches/calico-node.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  value:
+    name: IP6
+    value: "autodetect"
+- op: add
+  path: "/spec/template/spec/containers/0/env/-"
+  value:
+    name: FELIX_IPV6SUPPORT
+    value: "true"
+
+
```
Comment (jsturtevant) [inline templates/addons/calico-ipv6/patches/calico-node.yaml:18]: extra line

File: Makefile [hunk 2]
```diff
@@ -386,6 +386,8 @@ generate-e2e-templates: $(KUSTOMIZE)
 .PHONY: generate-addons
 generate-addons:
 	$(KUSTOMIZE) build $(ADDONS_DIR)/metrics-server > $(ADDONS_DIR)/metrics-server/metrics-server.yaml
+	$(KUSTOMIZE) build $(ADDONS_DIR)/calico > $(ADDONS_DIR)/calico.yaml
+	$(KUSTOMIZE) build $(ADDONS_DIR)/calico-ipv6 > $(ADDONS_DIR)/calico-ipv6.yaml
```
Comment (jsturtevant) [inline Makefile:390]: This is cool to see!

**Review by jsturtevant — COMMENTED** [review]
The diffs where a little harder than I had hoped due to the whitespace but that should be better in the future since kustomize should help standardize the final output.

**Comment by jsturtevant** [comment]
We should also bump the Windows calico versions to `sigwindowstools/calico-node:v3.22.0-hostprocess`

https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/f6ac019929d04b465d969a2f69d59759d8122b57/templates/addons/windows/calico/calico.yaml#L166
https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/f6ac019929d04b465d969a2f69d59759d8122b57/templates/addons/windows/calico/calico.yaml#L208
https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/f6ac019929d04b465d969a2f69d59759d8122b57/templates/addons/windows/calico/calico.yaml#L235

**Comment by CecileRobertMichon** [comment]
tests passed 🎉 

going to squash and bump version in a separate commit

**Comment by CecileRobertMichon** [comment]
/hold

ipv6 test failing after updating version

**Comment by jsturtevant** [comment]
> Does it matter if Windows and Linux aren't on the same patch version?

It is likely a good idea to keep them on the same version.  

> Is it possible to create an image for calico host process with 3.20.3?

Built a package for this version.

> for some reason 3.22 breaks ipv6. All tests passed with 3.20 (the current version in main). I'm thinking of reverting the bump to 3.22 to unblock this for now and investigate why 3.22 isn't working separately

This makes sense. It will easier identify the difference between 3.20 and 3.22

**Comment by CecileRobertMichon** [comment]
reverted 1.22 bump and upgraded to 1.20.4 which just came out

/test pull-cluster-api-provider-azure-e2e-full

**Comment by CecileRobertMichon** [comment]
prow got stuck

/retest

**Comment by jackfrancis** [comment]
I think the commit history of the version bump and reversion is lost to me, but this idea is a huge lgtm. Since there appears to be Windows impact @jsturtevant can you assign to me after you've given a final lgtm and I'll follow along with a review after that?

**Comment by CecileRobertMichon** [comment]
/retest

2nd time running into the prow timeout on the full job with no apparent test failure

**Comment by CecileRobertMichon** [comment]
I'm going to say this is good to go based on the fact that the last full test run 8/9 tests passed. The one that failed was an AKS provisioning error, and AKS doesn't use the Calico CNI we maintain in this repo, and that the AKS spec passed separately in the exp test.

/approve

**Comment by jsturtevant** [comment]
this looks to have broken the conformance jobs. We ran it early in the cycle but didn't run with latest updates

/test pull-cluster-api-provider-azure-conformance

**Pull request merged**